.github/workflows/browser-tests-beta.yml

```diff
@@ -141,3 +147,106 @@ jobs:
         id: run-safari-tests
         run: npm run test:safari -- --safari-tp
         continue-on-error: true # Beta browsers may have bugs
+
+    outputs:
+      test-result: ${{ steps.run-safari-tests.outcome }}
+
+  notify:
+    runs-on: ubuntu-latest
+    name: Notify on Failure
+    needs: [chrome-beta, firefox-beta, safari-tp]
+    environment: matrix-notifications
+
+    # `always()` is needed because jobs with `needs` are skipped if any
+    # dependency fails - and we need it exactly in such cases.
+    if: >-
+      always() && (
+        needs.chrome-beta.outputs.test-result == 'failure' ||
+        needs.firefox-beta.outputs.test-result == 'failure' ||
+        needs.safari-tp.outputs.test-result == 'failure'
+      )
+    steps:
+      - name: Prepare failure message
+        id: prepare-message
+        run: |
+          FAILED_BROWSERS=()
+          if [ "${{ needs.chrome-beta.outputs.test-result }}" == "failure" ]; then
+            FAILED_BROWSERS+=("Chrome Beta")
+          fi
+          if [ "${{ needs.firefox-beta.outputs.test-result }}" == "failure" ]; then
+            FAILED_BROWSERS+=("Firefox Beta")
+          fi
+          if [ "${{ needs.safari-tp.outputs.test-result }}" == "failure" ]; then
+            FAILED_BROWSERS+=("Safari Technology Preview")
+          fi
+          # Join array elements with ", "
+          printf -v FAILED_BROWSERS_STR '%s, ' "${FAILED_BROWSERS[@]}"
+          FAILED_BROWSERS_STR="${FAILED_BROWSERS_STR%, }"
+          echo "failed_browsers=$FAILED_BROWSERS_STR" >> "$GITHUB_OUTPUT"
+
+      - name: Send Matrix notification
+        env:
+          MATRIX_USER: jquerybot
+          MATRIX_PASSWORD: ${{ secrets.MATRIX_PASSWORD }}
+          MATRIX_ROOM_ID: ${{ secrets.MATRIX_ROOM_ID }}
```

Comment:
These two additional secrets are needed in the new matrix-notifications environment.

BTW, GitHub doesn't allow to rename environments, so changing that later requires a few more steps than it should.